source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,359 @@
+are processed when a connection becomes available.
+
+If there are operations in the wait queue **and** the number of connections
+in the pool is less than the ``maxPoolSize``, the driver establishes connections
+to process queued operations, up to the ``maxPoolSize``.
+
+.. example::
+
+   Your ``minPoolSize`` is **4** and your ``maxPoolSize`` is **10**. In
+   response to a successful email campaign, thousands of users log in to
+   your application. The driver uses the four connections in the pool,
+   and places other operations in the wait queue.
+
+   In response to the wait queue growth, the driver establishes more
+   connections to process operations, ``maxConnecting`` at a time until the
+   connection pool reaches ``maxPoolSize`` connections.
+
+Operations that cause the driver to create a new connection will be processed
+on that new connection, once established. If the driver returns a connection
+to the connection to the pool during connection creation, the returned connection
+is available for other operations in the wait queue.
+
+Reduction
++++++++++
+
+Use ``maxIdleTimeMS`` to reduce the number of connections in the pool to
+reduce resource requirements both on the application and on the database
+machines. The driver has no default, but 30 minutes is a good starting point.
+
+Connection Pool Sizing
+----------------------
+
+.. _connection-pool-sizing-java:
+
+Properly sizing your connection pool is critical to your application's
+performance.
+
+The rough guideline for sizing the connection pool is to set the ``minPoolSize``
+to the number of available CPU cores, and to set the ``maxPoolSize`` to double
+the number of CPU cores plus spindles (if using HDDs).
+
+.. example::
+
+   Your application has 4 available cores, and the servers hosting your database
+   have 8 CPU cores. Beginning with a ``minPoolSize`` of 4 and a ``maxPoolSize`` of 8
+   is a good starting point for connection pool size.
+
+.. important:: Don't Oversaturate the CPU
+
+   Remember that each application server will form connections. If each
+   application server has 4 CPU cores and there are 2 application servers
+   each configured with a ``minPoolSize`` of 4, a database server with
+   4 CPU cores may already be close to it's maximum number of performant
+   connections.
+
+For fine tuning your connection pool size, consider the following factors:
+
+- CPU
+- Database Server Disk I/O
+
+CPU
++++
+
+A CPU can process one operation at a time per core. If your CPU has eight
+cores, it can process eight operations in parallel. More operations will
+result in the underlying operating system scheduling operations and
+"sharing time" between operations.
+
+.. example::
+
+   The driver is processing two queries at the same time, and the underlying
+   operating system is scheduling these operations on the same CPU core.
+
+   If each operations takes ~100 ms to process, executing these operations
```

Comment:
This is true for CPU-bound work, but often times the driver spends a majority of its time blocked on network I/O, so increasing the concurrency can increase performance. This is the main motivation behind using async drivers.